glyphs-reader/src/corner_components.rs

```diff
@@ -0,0 +1,602 @@
+                Alignment::OutStroke => outstroke_angle,
+                Alignment::InStroke => instroke_angle,
+                Alignment::Middle => (instroke_angle + outstroke_angle) / 2.0,
+                _ => 0.0,
+            };
+
+        // rotate the paths around the origin and align them
+        // so that the origin of the corner starts on the target node
+        //https://github.com/googlefonts/glyphsLib/blob/f90e4060/Lib/glyphsLib/filters/cornerComponents.py#L384
+        let xform = Affine::translate(target_pt.to_vec2()).pre_rotate(angle);
+        self.apply_transform(xform);
+
+        AlignmentState {
+            instroke_pt,
+            outstroke_pt,
+            correction,
+        }
+    }
+
+    //https://github.com/googlefonts/glyphsLib/blob/f90e4060/Lib/glyphsLib/filters/cornerComponents.py#L396
+    fn recompute_instroke_intersection_point(
+        &self,
+        path: &Path,
+        target_node_ix: usize,
+    ) -> Option<Point> {
+        // see ref above, this just treats it as a line
+        let first_seg_as_line = &self.corner_path.nodes.as_slice()[..2];
+        let first_seg_as_line = Line::new(first_seg_as_line[0].pt, first_seg_as_line[1].pt);
+        let instroke = path.get_previous_segment(target_node_ix).unwrap();
+        unbounded_seg_seg_intersection(first_seg_as_line.into(), instroke)
+    }
+
+    //https://github.com/googlefonts/glyphsLib/blob/f90e4060b/Lib/glyphsLib/filters/cornerComponents.py#L401
+    fn recompute_outstroke_intersection_point(
+        &self,
+        original_outstroke: PathSeg,
+        hint: &Hint,
+    ) -> Option<Point> {
+        if hint.is_flipped() {
+            unbounded_seg_seg_intersection(
+                self.corner_path
+                    .get_previous_segment(self.corner_path.nodes.len() - 1)
+                    .unwrap(),
+                original_outstroke,
+            )
+        } else {
+            // the python all uses custom geometry fns, which i would like to avoid..
+            let nearest = original_outstroke.nearest(self.last_point(), 1e-6);
+            Some(original_outstroke.eval(nearest.t))
+        }
+    }
+
+    fn stretch_first_seg_to_fit(&mut self, intersection_pt: Point) {
+        let delta = intersection_pt - self.corner_path.nodes[0].pt;
+        self.corner_path.nodes[1].pt += delta;
+    }
+}
+
+/// Find the intersection of two unbounded segments
+///
+/// <https://github.com/googlefonts/glyphsLib/blob/f90e4060b/Lib/glyphsLib/filters/cornerComponents.py#L127>
+fn unbounded_seg_seg_intersection(seg1: PathSeg, seg2: PathSeg) -> Option<Point> {
+    // Line-line intersection
+    match (seg1, seg2) {
+        (PathSeg::Line(one), PathSeg::Line(two)) => one.crossing_point(two),
+        (seg, PathSeg::Line(line)) | (PathSeg::Line(line), seg) => {
+            // a value by which we extend our line, to find the crossing point.
+            // should be enough for anybody!
+            const LITERALLY_UNBOUNDED: f64 = 10000.;
```

Comment:
depends on the font's units per EM... Hopefully this won't bite us later.
The python code doesn't need this magic constant. Why do we?